File: alpaca/templates/python/lib/http_client/response_handler.py
import string

# ResponseHandler takes care of decoding the response body into suitable type
class ResponseHandler():

	@staticmethod
	def get_body(response):
		typ = response.headers.get('content-type')
		body = response.text
{{if .Api.response.formats.json}}
		# Response body is in JSON
		if string.find(typ, 'json') != -1:
			body = response.json()
{{end}}
		return body

File: templates/node/lib/http_client/response_handler.js
var response = module.exports;

/**
 * This module takes care of decoding the response body into suitable type
 */
response.getBody = function(res, body, callback) {
  var type = res.headers['content-type'], error = null;
{{if .Api.Response.Formats.Json}}
  // Response body is in JSON
  if (type.indexOf('json') != -1 && typeof(body) != 'object') {
    try {
      body = JSON.parse(body || '{}');
    } catch (err) {
      error = err;
    }
  }
{{end}}
  return callback(error, res, body);
};
